File: knowledge-base-backend-master/kb-user-auth/src/main/java/com/knowledge/base/userauth/service/RoleService.java
package com.knowledge.base.userauth.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.knowledge.base.userauth.dto.RoleDTO;
import com.knowledge.base.userauth.vo.RoleVO;

import java.util.List;

/**
 * Role Service interface
 *
 * <p>Provides role-related business logic</p>
 *
 * @author airwzz999
 * @since 1.0.0
 */
public interface RoleService {

    /**
     * Create a role
     *
     * @param roleDTO role information
     * @return role ID
     */
    Long createRole(RoleDTO roleDTO);

    /**
     * Update a role
     *
     * @param roleDTO role information
     * @return whether successful
     */
    Boolean updateRole(RoleDTO roleDTO);

    /**
     * Delete a role
     *
     * @param roleId role ID
     * @return whether successful
     */
    Boolean deleteRole(Long roleId);

    /**
     * Query a role by ID
     *
     * @param roleId role ID
     * @return role information
     */
    RoleVO getRoleById(Long roleId);

    /**
     * Paginated role query
     *
     * @param current current page
     * @param size    page size
     * @param keyword search keyword
     * @return paginated role information
     */
    IPage<RoleVO> pageRoles(Long current, Long size, String keyword);

    /**
     * Get all roles
     *
     * @return role list
     */
    List<RoleVO> getAllRoles();

    /**
     * Assign permissions
     *
     * @param roleId        role ID
     * @param permissionIds permission ID list
     * @return whether successful
     */
    Boolean assignPermissions(Long roleId, List<Long> permissionIds);

    /**
     * Get a role's permissions
     *
     * @param roleId role ID
     * @return permission ID list
     */
    List<Long> getRolePermissions(Long roleId);
}

File: knowledge-base-frontend-master/src/services/role.service.ts
import { http } from './request';
import { Role, Permission } from '@/types';

export const roleService = {
  // Get role list
  getRoles: (config?: any) => http.get<Role[]>('/auth/roles/list', config),

  // Get role details
  getRole: (id: string) => {
    return http.get<Role>(`/auth/roles/${id}`);
  },

  // Create role
  createRole: (data: { name: string; code: string; description?: string; permissions: string[] }) => {
    const { permissions: _permissions, ...payload } = data;
    return http.post<string | number>('/auth/roles', payload);
  },

  // Update role
  updateRole: (
    id: string,
    data: { name?: string; code?: string; description?: string; permissions?: string[] }
  ) => {
    const { permissions: _permissions, ...payload } = data;
    return http.put<boolean>('/auth/roles', { id: String(id), ...payload });
  },

  // Delete role
  deleteRole: (id: string) => http.delete<boolean>(`/auth/roles/${id}`),

  // Get all permissions
  getPermissions: () => http.get<Permission[]>('/auth/permissions'),

  // Get role permissions
  getRolePermissions: (roleId: string, config?: any) =>
    http.get<Array<string | number>>(`/auth/roles/${roleId}/permissions`, config).then((res) =>
      (res || []).map((id) => String(id))
    ),

  // Assign role permissions
  assignPermissions: (roleId: string, permissionIds: string[], config?: any) =>
    http.post<boolean>(
      `/auth/roles/${roleId}/permissions`,
      permissionIds.map((id) => String(id)),
      config
    ),
};

export default roleService;
